frontend/pages/policies/ManagePoliciesPage/components/PoliciesPaginatedList/PoliciesPaginatedList.tsx

```diff
@@ -0,0 +1,198 @@
+  renderItemRow?: (
+    item: IFormPolicy,
+    onChange: (item: IFormPolicy) => void
+  ) => ReactElement | false | null | undefined;
+  onToggleItem: (item: IFormPolicy) => IFormPolicy;
+  onCancel: () => void;
+  onSubmit: (formData: IFormPolicy[]) => void;
+  isUpdating: boolean;
+  disableSave?: (changedItems: IFormPolicy[]) => boolean | string;
+  teamId: number;
+  footer: ReactElement | undefined | null;
+}
+
+const baseClass = "policies-paginated-list";
+
+function PoliciesPaginatedList(
+  {
+    isSelected,
+    renderItemRow,
+    onToggleItem,
+    onCancel,
+    onSubmit,
+    isUpdating,
+    disableSave,
+    teamId,
+    footer,
+  }: IPoliciesPaginatedListProps,
+  ref: Ref<IPaginatedListHandle<IFormPolicy>>
+) {
+  // Create a ref to use with the PaginatedList, so we can access its dirty items.
+  const paginatedListRef = useRef<IPaginatedListHandle<IFormPolicy>>(null);
+
+  const [saveDisabled, setSaveDisabled] = useState<string | boolean>(false);
+
+  // Allow parents access to the `getDirtyItems` of the underlying PaginatedList.
+  useImperativeHandle(ref, () => ({
+    getDirtyItems() {
+      if (paginatedListRef.current) {
+        return paginatedListRef.current.getDirtyItems();
+      }
+      return [];
+    },
+  }));
+
+  // When "save" is clicked, call the parent's `onSubmit` with the set of changed items.
+  const onClickSave = () => {
+    let changedItems: IFormPolicy[] = [];
+    if (paginatedListRef.current) {
+      changedItems = paginatedListRef.current.getDirtyItems();
+    }
+    onSubmit(changedItems);
+  };
+
+  // When any item in the list is changed, check whether we should disable the save button.
+  const onUpdate = useCallback(
+    (changedItems: IFormPolicy[]) => {
+      if (!disableSave) {
+        return;
+      }
+      setSaveDisabled(disableSave(changedItems));
+    },
+    [disableSave]
+  );
+
+  // Fetch a single page of policies.
+  const queryClient = useQueryClient();
+  const DEFAULT_PAGE_SIZE = 8;
+  const DEFAULT_SORT_COLUMN = "name";
+
+  const fetchPage = useCallback((pageNumber: number) => {
+    let fetchPromise;
+
+    if (teamId === APP_CONTEXT_ALL_TEAMS_ID) {
+      fetchPromise = queryClient.fetchQuery(
+        [
+          {
+            scope: "globalPolicies",
+            page: pageNumber,
+            perPage: DEFAULT_PAGE_SIZE,
+            query: "",
+            orderDirection: "asc" as const,
+            orderKey: DEFAULT_SORT_COLUMN,
+          },
+        ],
+        ({ queryKey }) => {
+          return globalPoliciesAPI.loadAllNew(queryKey[0]);
+        }
+      );
+    } else {
+      fetchPromise = queryClient.fetchQuery(
+        [
+          {
+            scope: "teamPolicies",
+            page: pageNumber,
+            perPage: DEFAULT_PAGE_SIZE,
+            query: "",
+            orderDirection: "asc" as const,
+            orderKey: DEFAULT_SORT_COLUMN,
+            teamId,
+            mergeInherited: false,
```

Comment:
Do we want to set this conditionally to allow merging inherited policies for the Other workflows modal?